src/validation/custom.validation.ts

```diff
@@ -78,11 +78,75 @@ export const checkDateValueIsValid = (errMsg: string, dayStr: string = "", month
   if ( dayStr !== "" || monthStr !== "" || yearStr !== "" ) {
-    checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
-    checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+    const areDateFieldsPresent = checkDateFieldsPresent(dayStr, monthStr, yearStr);
+    if (areDateFieldsPresent) {
+      const isDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
+      if (isDateValid) {
+        const isDateInThePast = checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+        if (isDateInThePast) {
+          checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
+        }
+      }
+    }
+  }
+  return true;
+};
+
+export const checkIdentityDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  const isStartDateFields = checkStartDate(dayStr, monthStr, yearStr);
+  if (isStartDateFields) {
     checkDateIsWithinLast3Months(ErrorMessages.IDENTITY_CHECK_DATE_NOT_WITHIN_PAST_3_MONTHS, dayStr, monthStr, yearStr);
   }
+  return true;
+};
 
+export const checkStartDate = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  const isDatePresent = checkDateIsNotCompletelyEmpty(ErrorMessages.ENTER_DATE, dayStr, monthStr, yearStr);
+  if (isDatePresent) {
+    const areDateFieldsPresent = checkDateFieldsPresent(dayStr, monthStr, yearStr);
+    if (areDateFieldsPresent) {
+      const isDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE, dayStr, monthStr, yearStr);
+      if (isDateValid) {
+        checkDateIsInPastOrToday(ErrorMessages.DATE_NOT_IN_PAST_OR_TODAY, dayStr, monthStr, yearStr);
+      }
+    }
+  }
+  return true;
+};
+
+export const checkDateFieldsPresent = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  if (dayStr === "") {
+    throw new Error(ErrorMessages.DAY);
+  } else if (monthStr === "") {
+    throw new Error(ErrorMessages.MONTH);
+  } else if (yearStr === "") {
+    throw new Error(ErrorMessages.YEAR);
+  }
+  return true;
+};
+
+export const checkDateOfBirth = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  const isDatePresent =  checkDateIsNotCompletelyEmpty(ErrorMessages.ENTER_DATE_OF_BIRTH, dayStr, monthStr, yearStr);
+  if (isDatePresent) {
+    const areDateOfBirthFieldsPresent = checkDateOfBirthFieldsPresent(dayStr, monthStr, yearStr);
+    if (areDateOfBirthFieldsPresent) {
+      const isDateValid = checkDateValueIsValid(ErrorMessages.INVALID_DATE_OF_BIRTH, dayStr, monthStr, yearStr);
+      if (isDateValid) {
+        checkDateIsInPast(ErrorMessages.DATE_NOT_IN_PAST, dayStr, monthStr, yearStr);
+      }
+    }
+  }
+  return true;
+};
+
+export const checkDateOfBirthFieldsPresent = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
```

Comment:
Suggest renaming to `checkDateOfBirthFieldsArePresent`.

Reply:
Renamed as above